### PR TITLE
Assertions and logs for server side transaction streaming

### PR DESCRIPTION
**Pull request by martinfurmanski**

### CODE REVIEW

File: enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/tx/ChunkedTransactionStream.java [hunk 1]
```diff
@@ -96,6 +98,11 @@ else if ( txCursor.next() )
 
             CommittedTransactionRepresentation tx = txCursor.get();
             lastTxId = tx.getCommitEntry().getTxId();
+            if ( lastTxId != expectedTxId )
+            {
+                throw new IllegalStateException( "Transaction cursor out of order." );
```
Comment (hugofirth) [inline enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/tx/ChunkedTransactionStream.java]: May be worth providing the expected and actual values of the transaction cursor to help with debugging from the logs?

Comment (martinfurmanski) [reply enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/tx/ChunkedTransactionStream.java]: Yes, will fix!

**Comment by martinfurmanski** [comment]
@hugofirth Updated the printout to include the expected and actual values. Also fixed the failing test. PTAL (please take another look)!

**Review by hugofirth — APPROVED** [review]
All looks good to me - nothing specific to add.

**Pull request merged**